Support sbt 1.4.0 when COURSIER_CACHE is overridden

anilkumarmyla opened the pull request:
Fixes #74 

When COURSIER_CACHE is overriden (say to `.cache/coursier/v1`), in sbt-1.4.0 we would get file path as `${BASE}/.cache/coursier/v1/https/repo1.maven.org/maven2/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar` in which case we need to interpolate ${BASE} as well. This PR attempts to do so

cb372 commented:
Do you want to try adding a scripted test for this? I think adding the following line to build.sbt in the test project should reproduce the issue:

```
ThisBuild / csrCacheDirectory := (ThisBuild / baseDirectory).value / "coursier-cache"
```

When I upgraded the example project to sbt 1.4.0 and added that line I got debug output like this:

```
[debug] Source to library relations:
[debug]   ${BASE}/src/main/scala/foo/MyCaseClass.scala -> ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.11/scala-library-2.12.11.jar
[debug]   ${BASE}/src/main/scala/foo/MyCaseClass.scala -> ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/typelevel/cats-effect_2.12/0.10.1/cats-effect_2.12-0.10.1.jar
[debug]   ${BASE}/src/main/scala/foo/Example.scala -> ${BASE}/coursier-cache/https/repo1.maven.org/maven2/com/chuusai/shapeless_2.12/2.3.3/shapeless_2.12-2.3.3.jar
[debug]   ${BASE}/src/main/scala/foo/Example.scala -> ${BASE}/coursier-cache/https/repo1.maven.org/maven2/com/google/guava/guava/26.0-jre/guava-26.0-jre.jar
[debug]   ${BASE}/src/main/scala/foo/Example.scala -> ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/typelevel/cats-effect_2.12/0.10.1/cats-effect_2.12-0.10.1.jar
[debug]   ${BASE}/src/main/scala/foo/Example.scala -> ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.11/scala-library-2.12.11.jar
[debug]   ${BASE}/src/main/scala/foo/Example.scala -> ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/typelevel/cats-core_2.12/1.2.0/cats-core_2.12-1.2.0.jar
[debug]   ${BASE}/src/main/scala/foo/Example.scala -> ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/http4s/http4s-blaze-server_2.12/0.18.16/http4s-blaze-server_2.12-0.18.16.jar
[debug] Library dependencies:
[debug]   ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/typelevel/cats-effect_2.12/0.10.1/cats-effect_2.12-0.10.1.jar
[debug]   ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/http4s/http4s-blaze-server_2.12/0.18.16/http4s-blaze-server_2.12-0.18.16.jar
[debug]   ${BASE}/coursier-cache/https/repo1.maven.org/maven2/com/chuusai/shapeless_2.12/2.3.3/shapeless_2.12-2.3.3.jar
[debug]   ${BASE}/coursier-cache/https/repo1.maven.org/maven2/com/google/guava/guava/26.0-jre/guava-26.0-jre.jar
[debug]   ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/typelevel/cats-core_2.12/1.2.0/cats-core_2.12-1.2.0.jar
[debug]   ${BASE}/coursier-cache/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.11/scala-library-2.12.11.jar
```

anilkumarmyla commented:
Added the scripted test. I was wondering how I could add an env variable while doing scripted test and completely forgot I could override it in build.sbt. Thanks for the suggestion, added the test and verified it fails without this change and passes with the change

anilkumarmyla commented:
Hmm, I realized this key doesn't exist in sbt 1.2.x or lower and fails the test for those builds 😬 Any suggestions on how I could resolve? Should I pin the scripted test to sbt 1.4.0?

cb372 commented:
I can't think of a good way to make just this test run only on certain versions of sbt 1.x.

If you want to give up on writing a test for this, I'm fine with that. You've already tested it manually and I've reviewed the code, so I'm satisfied that it works. In an ideal world we'd have an automated test, but I don't think it's worth spending too much effort on.

anilkumarmyla commented:
Yeah we can add a test when we stop supporting sbt 1.2.x or lower, for now i've reverted the test commit
